langchain_mcp_adapters/tools.py

```diff
@@ -219,20 +219,25 @@ def to_fastmcp(tool: BaseTool) -> FastMCPTool:
         TypeError: If the tool's args_schema is not a BaseModel subclass.
         NotImplementedError: If the tool has injected arguments.
     """
-    if not issubclass(tool.args_schema, BaseModel):
+    if (
+        tool.args_schema is None
+        or isinstance(tool.args_schema, dict)
+        or not issubclass(tool.args_schema, BaseModel)
+    ):
         msg = (
             "Tool args_schema must be a subclass of pydantic.BaseModel. "
             "Tools with dict args schema are not supported."
         )
         raise TypeError(msg)
 
-    parameters = tool.tool_call_schema.model_json_schema()
+    args_schema = tool.args_schema
```

Comment:
args_schema vs. tool_call_schema -- just checking quickly were there any other fixes here that we needed?